src/masternode.cpp

```diff
@@ -749,84 +749,81 @@ bool CMasternodePing::VerifySignature(CPubKey& pubKeyMasternode, int &nDos) {
         nDos = 1;
         return false;
     }
 
-    if(fCheckSigTimeOnly) {
+    if (fCheckSigTimeOnly) {
         CMasternode* pmn = mnodeman.Find(vin);
-        if(pmn) return VerifySignature(pmn->pubkey2, nDos);
+        if (pmn) return VerifySignature(pmn->pubkey2, nDos);
         return true;
     }
 
-    LogPrint("masternode", "CMasternodePing::CheckAndUpdate - New Ping %s - %s %s %d\n", vin.ToString(), GetHash().ToString(), blockHash.ToString(), sigTime);
+    LogPrint("masternode", "CMasternodePing::CheckAndUpdate -- New ping: masternode=%s  blockHash=%s  sigTime=%d\n", vin.prevout.ToStringShort(), blockHash.ToString(), sigTime);
 
     // see if we have this Masternode
     CMasternode* pmn = mnodeman.Find(vin);
-    if(pmn != NULL && pmn->protocolVersion >= mnpayments.GetMinMasternodePaymentsProto())
-    {
-        if (fRequireEnabled && !pmn->IsEnabled() && !pmn->IsPreEnabled()) return false;
 
-        // LogPrintf("mnping - Found corresponding mn for vin: %s\n", vin.ToString());
-        // update only if there is no known ping for this masternode or
-        // last ping was more then MASTERNODE_MIN_MNP_SECONDS-60 ago comparing to this one
-        if(!pmn->IsPingedWithin(MASTERNODE_MIN_MNP_SECONDS - 60, sigTime))
-        {
-            if(!VerifySignature(pmn->pubkey2, nDos))
-                return false;
+    if (pmn == NULL || pmn->protocolVersion < mnpayments.GetMinMasternodePaymentsProto()) {
+        LogPrint("masternode", "CMasternodePing::CheckAndUpdate -- Couldn't find compatible Masternode entry, masternode=%s\n", vin.prevout.ToStringShort());
+        return false;
+    }
 
-            {
-                LOCK(cs_main);
-                BlockMap::iterator mi = mapBlockIndex.find(blockHash);
-                if (mi != mapBlockIndex.end() && (*mi).second)
-                {
-                    if((*mi).second->nHeight < chainActive.Height() - 24)
-                    {
-                        LogPrintf("CMasternodePing::CheckAndUpdate - Masternode %s block hash %s is too old\n", vin.ToString(), blockHash.ToString());
-                        // Do nothing here (no Masternode update, no mnping relay)
-                        // Let this node to be visible but fail to accept mnping
-
-                        return false;
-                    }
-                } else {
-                    if (fDebug) LogPrintf("CMasternodePing::CheckAndUpdate - Masternode %s block hash %s is unknown\n", vin.ToString(), blockHash.ToString());
-                    // maybe we stuck so we shouldn't ban this node, just fail to accept it
-                    // TODO: or should we also request this block?
+    if (fRequireEnabled && !pmn->IsEnabled() && !pmn->IsPreEnabled()) return false;
 
-                    return false;
-                }
-            }
-            pmn->lastPing = *this;
```

Comment:
it's over there https://github.com/dashpay/dash/pull/1020/files/d1aa36adc4a1165952f1051a043bb910813cb2ae#diff-031d4f9d201291e57fa6adbede2755a1R811 now :)